src/Npgsql/Replication/NpgsqlTimelineHistoryFile.cs

```diff
@@ -0,0 +1,32 @@
+﻿using JetBrains.Annotations;
+
+namespace Npgsql.Replication
+{
+    /// <summary>
+    /// Represents a PostgreSQL timeline history file
+    /// </summary>
+    public readonly struct NpgsqlTimelineHistoryFile
+    {
+        internal  NpgsqlTimelineHistoryFile(string fileName, byte[] content)
+        {
+            FileName = fileName;
+            Content = content;
+        }
+
+        /// <summary>
+        /// File name of the timeline history file, e.g., 00000002.history.
+        /// </summary>
+        public string FileName { get; }
+
+
+        // While it is pretty safe to assume that a timeline history file
+        // only contains ASCII bytes since it is automatically written and
+        // parsed by the PostgreSQL backend, we don't want to claim anything
+        // about its content (we get it as bytes and we hand it over as bytes).
+
+        /// <summary>
+        /// Contents of the timeline history file.
+        /// </summary>
+        public byte[] Content { get; }
```

Comment:
Are these timeline history files big? If so, does it make sense for us to stream them?

Reply:
No they aren't.
Typical content looks like the following (with one line added per timeline switch):

```
1	0/30192E8	no recovery target specified

```

So it's around 40 bytes per timeline switch.

Reply:
Makes sense!